added comments to the phytoplankton mortality section

charliestock opened the pull request:
This pull request improves the commenting in the phytoplankton loss section.  There were no changes to the actual code and it should not change answers.  I have tagged @jessluo and @gabyneg for a quick look to ensure the commenting is accessible and informative.

charliestock committed:
added comments to the phytoplankton mortality section

gabyneg commented on generic_tracers/generic_COBALT.F90:
```diff
@@ -3534,40 +3534,78 @@ subroutine generic_COBALT_update_from_source(tracer_list,Temp,Salt,rho_dzt,dzt,h
 
     do k = 1, nk ; do j = jsc, jec ; do i = isc, iec; !{
 
+       ! 3.2.1 Calculate losses of phytoplankton to aggregation and mortality and the rate direct sinking.
```
minor comment, add "of" before "direct sinking"

gabyneg reviewed (approved):
I think this looks great! I just added a tiny comment for clarity, but aside from that I think it looks great!

charliestock committed:
finished commenting and cleanup of plankton foodweb calculations

charliestock commented:
Hello @yichengt900 @gabyneg @jessluo - I added a commit that addressed the comment typo that Gaby caught and added comments to the rest of the plankton foodweb section.  Take a quick look when/if you have a chance and perhaps we can pull this in on Monday?

thanks,
Charlie

yichengt900 commented on generic_tracers/generic_COBALT.F90:
```diff
@@ -3534,40 +3534,78 @@ subroutine generic_COBALT_update_from_source(tracer_list,Temp,Salt,rho_dzt,dzt,h
+       ! 3.2.1 Calculate losses of phytoplankton to aggregation and mortality and the rate of direct sinking.
        !
-       ! 3.2.1 Calculate losses of phytoplankton to aggregation and mortality
+       ! These losses depend on whether phytoplankton are growing well or "stressed". Stress is quanitified as a
```
"quantified"

yichengt900 commented on generic_tracers/generic_COBALT.F90:
```diff
@@ -3534,40 +3534,78 @@ subroutine generic_COBALT_update_from_source(tracer_list,Temp,Salt,rho_dzt,dzt,h
+       ! growth_ratio = min( max(mu_mem,0)/(frac_mu_stress*P_C_max(T)), 1.0)
+       !
+       ! Note that growth_ratio = 1 when mu_mem >= frac_mu_stress*P_C_max(T) and approaches 0 as mu_mem -> 0.  The
+       ! stress factor is then claculated as:
```
"calculated"

yichengt900 commented on generic_tracers/generic_COBALT.F90:
```diff
@@ -3596,19 +3640,21 @@ subroutine generic_COBALT_update_from_source(tracer_list,Temp,Salt,rho_dzt,dzt,h
+    ! Assume that individually sinking phytoplankton, which sink at slow rates relative to aggregates and fecal
+    ! pellets, collect in a nepholoid layer and are available for resuspension if they are exposed to mixing. This
+    ! is accomplished by setting the vertical sinking rate in the bottom layer to 0, and is assumed to occur when 
+    ! when the depth is less than twice the depth of active mixing.  Cells are otherwise assumed to sink into the
```
Remove extra "when"

yichengt900 reviewed (commented):
@charliestock , Thanks for adding those comments. Overall they look good to me except a few typos. I can fix them on my end.

yichengt900 committed:
fix typos

jessluo commented on generic_tracers/generic_COBALT.F90:
```diff
@@ -3579,9 +3617,15 @@ subroutine generic_COBALT_update_from_source(tracer_list,Temp,Salt,rho_dzt,dzt,h
-       !
        ! 3.2.3 Calculate losses to exudation
        !
+       ! Phytoplankton are assumed to lose a constant fraction of carbon they fix to dissolved organic nutrients 
```
should this be nitrogen here as the model units are in nitrogen? and also because in line 3624 the comment references nitrogen.

charliestock replied:
I think a change to nitrogen would be fine.  I think I put C because, if I recall correctly, that is what Baines and Pace used, but N is indeed more precise for COBALT.

jessluo commented on generic_tracers/generic_COBALT.F90:
```diff
@@ -3596,19 +3640,21 @@ subroutine generic_COBALT_update_from_source(tracer_list,Temp,Salt,rho_dzt,dzt,h
-       ! assume that individually sinking phytoplankton collect in nepholoid layer
-       ! and are available for resuspension if they are exposed to mixing
        do n = 1,NUM_PHYTO
          if (cobalt%zt(i,j,nk).le.(2.0*hblt_depth(i,j))) then
```
I'm curious about this -- if you use just the `hblt_depth` here instead of twice the depth of active mixing, does this result in too much burial and removal of phytoplankton N/Si/Fe?

charliestock replied:
Hi Jessica - right now burial is turned off since we were very concerned with too much burial in regional applications with enhanced resolution of nearshore areas.   We also found, however, that assuming that slowly sinking cells were immediately buried and remineralized in the nearshore regions rather than being available for re-entrainment was suppressing nearshore chlorophyll.  

I think these will be a good topic for additional sensitivity, with the Mississippi hypoxic zone likely providing a good test case.

jessluo commented on generic_tracers/generic_COBALT.F90:
```diff
@@ -3627,20 +3673,25 @@ subroutine generic_COBALT_update_from_source(tracer_list,Temp,Salt,rho_dzt,dzt,h
-       !
-       ! Production of detritus and dissolved organic material from zooplankton egestion
-       !
+       ! Note: For zooplankton ingestion, the "assimilation efficieny" is determined by 1.0 - the egested fraction.
```
should be "efficiency"

jessluo commented on generic_tracers/generic_COBALT.F90:
```diff
@@ -3741,79 +3774,84 @@ subroutine generic_COBALT_update_from_source(tracer_list,Temp,Salt,rho_dzt,dzt,h
-          ! is negative, zooplankton are lost to large detritus
-          !
+          ! Ingested material that does not go to zooplankton production or egestion (i.e., detrital production or
+          ! production of dissolved organic material is excreted as nh4 or po4 as part of the respiration process.
```
missing parentheses after "material"

jessluo reviewed (approved):
looks good! caught a few more typos.

yichengt900 committed:
fix more typos

yichengt900 commented:
Thanks @jessluo and @gabyneg! @charliestock, I fixed all the typos except for two great questions from @jessluo. I will merge this PR once they are resolved

jessluo commented:
> Thanks @jessluo and @gabyneg! @charliestock, I fixed all the typos except for two great questions from @jessluo. I will merge this PR once they are resolved

The second question is more of a science question, so I think only addressing the question on line 3622 would be fine for this PR.

charliestock commented:
Thanks for the questions and comment typo fixes (and sorry there were so many).  Agree that the second question is a good science one that can be addressed through sensitivities and subsequent pull requests.  @yichengt900, if you could switch the other to nitrogen, that would be great.

yichengt900 committed:
Update generic_COBALT.F90

yichengt900 commented:
Thanks again @charliestock, @jessluo and @gabyneg for all your contributions to this PR. @charliestock I addressed all of the comments and this PR is ready to merge.
